website/docusaurus.config.js

```diff
@@ -9,6 +9,57 @@ const util = require("util");
 const webpack = require("webpack");
 const execAsync = util.promisify(require("child_process").exec);
 
+/**
+ * Modify the svgo configuration (in place) to prevent it from minifying IDs in SVGs
+ *
+ * Refs:
+ * - https://github.com/facebook/docusaurus/issues/8297
+ * - https://github.com/svg/svgo/issues/1714
+ * - https://linear.app/foxglove/issue/FG-7251/logos-are-cut-off-on-mcapdev
+ *
+ * @param {webpack.Configuration} config
+ */
+function modifySvgoConfig(config) {
+  const NEW_SVGO_CONFIG = {
+    plugins: [
+      {
+        name: "preset-default",
+        params: {
+          overrides: {
+            removeTitle: false,
+            removeViewBox: false,
+            cleanupIDs: false, // do not change IDs
+          },
+        },
+      },
+    ],
+  };
+  // find the svgo config rule and replace it
+  if (config.module?.rules instanceof Array) {
+    for (const rule of config.module.rules) {
+      if (typeof rule === "object" && rule.test?.toString() === "/\\.svg$/i") {
+        if (rule.oneOf instanceof Array) {
+          for (const nestedRule of rule.oneOf) {
+            if (nestedRule.use instanceof Array) {
+              for (const loader of nestedRule.use) {
+                if (
```

Comment:
This level of nesting is really intense... at minimum it would be nice to flip these conditions around and use early returns.

Alternatively this version looks a bit cleaner:
https://github.com/facebook/docusaurus/issues/8297#issuecomment-1323343137

I imagine you could use a `try/catch` around some of the modifications to reduce the number of things that need to be checked up front and just silently continue in the event of an error.

Reply:
The conditionals are more for the linter than actually being used because the webpack config is fairly static. This was really instead of using a lot of `ts-ignore` comments. (I tried something like the linked version and it required a lot of them.)

I'll play around with it a bit more, but I didn't want to spend too much time fighting the linter over webpack config type definitions :)